src/main/java/ru/practicum/shareit/item/ItemServiceImpl.java

```diff
@@ -2,86 +2,167 @@
         }
         Item newItem = new Item(
                 itemId,
-                itemModifyDto.getName() != null ? itemModifyDto.getName() : item.getName(),
-                itemModifyDto.getDescription() != null ? itemModifyDto.getDescription() : item.getDescription(),
-                itemModifyDto.getAvailable() != null ? itemModifyDto.getAvailable() : item.isAvailable(),
+                requestItemDto.getName() != null ? requestItemDto.getName() : item.getName(),
+                requestItemDto.getDescription() != null ? requestItemDto.getDescription() : item.getDescription(),
+                requestItemDto.getAvailable() != null ? requestItemDto.getAvailable() : item.isAvailable(),
                 item.getOwner(),
                 item.getRequest());
-        return ItemMapper.toItemDto(itemRepository.updateItem(newItem));
+        return itemMapper.toItemDto(itemRepository.save(newItem));
     }
 
     @Override
+    @Transactional
     public void deleteItem(Long userId, Long itemId) {
         User user = findUserById(userId);
-        Item item = getItemById(itemId);
+        Item item = findItemById(itemId);
         if (!item.getOwner().getId().equals(user.getId())) {
             throw new CRUDException("Удаление доступно только для своих вещей");
         }
-        itemRepository.deleteItem(item);
+        itemRepository.delete(item);
     }
 
-    private Item getItemById(Long itemId) {
-        return itemRepository.getItem(itemId)
+    @Override
+    @Transactional
+    public CommentDto createComment(Long userId, Long itemId, RequestCommentDto requestCommentDto) {
+        User user = findUserById(userId);
+        Item item = findItemById(itemId);
+        bookingRepository.findPastBooking(user.getId(), item.getId())
+                .orElseThrow(() -> new ChekParamException(
+                        String.format("У пользователя с id = %d не найдено завершенное бронирование вещи с id = %d",
+                                user.getId(),
+                                item.getId())));
+        Comment comment = Comment.builder()
+                .text(requestCommentDto.getText())
+                .item(item)
+                .author(user)
+                .created(LocalDateTime.now())
+                .build();
+        return commentMapper.toCommentDto(commentRepository.save(comment));
+    }
+
+    private Item findItemById(Long itemId) {
+        return itemRepository.findById(itemId)
                 .orElseThrow(() -> new NotFoundException(String.format("Вещь с id = %d не найден", itemId)));
     }
 
     private User findUserById(Long userId) {
-        return userRepository.findUserById(userId)
+        return userRepository.findById(userId)
                 .orElseThrow(() -> new NotFoundException(String.format("Пользователь с id = %d не найден", userId)));
     }
+
+    private Map<Long, ShrotBookingDto> getLastNextBookings(Long userId, List<Long> itemIds, BookingType type) {
```

Comment:
Опечатка в имени класса
```suggestion
    private Map<Long, ShortBookingDto> getLastNextBookings(Long userId, List<Long> itemIds, BookingType type) {
```